ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/State.hs

```diff
@@ -168,7 +168,40 @@ epochInfoPrecomputedTransitionInfo shape transition st =
   Extending
 -------------------------------------------------------------------------------}
 
--- | Extend the telescope until the specified slot is within the era at the tip
+-- | Extend the telescope until the specified slot is within the era at the tip.
+--
+-- Note that transitioning to a later era might create new values in the ledger
+-- tables (like Byron -> Shelley does in Cardano). Therefore the result of this
+-- function is a @DiffMK@.
+--
+-- If we are crossing no era boundaries, this whole function is a no-op that
+-- only creates an empty @DiffMK@, because the @Telescope.extend@ function will
+-- do nothing.
+--
+-- If we are crossing one era boundary, the ledger tables might be populated
+-- with whatever @translateLedgerStateWith@ returns.
+--
+-- If we are crossing multiple era boundaries, the diffs generated when crossing
+-- an era boundary will be prepended to the ones produced by later era
+-- boundaries and, in order to all match the resulting era, they will be
+-- translated to later eras.
+--
+-- This means in particular that if for example in Cardano we extend from Byron
+-- to Mary, we will:
+--
+-- 1. translate the ledger state from Byron to Shelley, which produces a Shelley
+--    ledger state together with a big set of Shelley differences as ledger
+--    tables are populated only from Shelley onwards.
+--
+-- 2. keep the Shelley diffs aside, and translate the Shelley ledger state
+--    without ledger tables, which produces a Mary ledger state together with a
+--    set of Mary differences.
+--
+-- 3. Translate the Shelley diffs to Mary differences, and prepend them to the
+--    ones created in the step 2.
+--
+-- 4. Attach the diffs resulting from step 3 to the Mary ledger state from step
+--    2, and return it.
```

Comment:
I'm unsure how much we should talk about `Cardano` eras here. Maybe we can rewrite this example to mention generic `era1`, `era2`, `era3` or something?

Reply:
Sounds reasonable